docs/reference-guide/modules/commands/pages/infrastructure.adoc

```diff
@@ -1,252 +1,177 @@
- Exceptions and timeouts declared on the method are ignored.
-* Any other return type will cause the method to block until a result is available.
- The result is cast to the return type (causing a `ClassCastException` if the types do not match).
+[source,java]
+----
+public interface ManagedEntity<ID, E> {
+    ID identifier();           // Get the entity identifier
+    E entity();               // Get the current entity state
+    E applyStateChange(UnaryOperator<E> change);  // Modify entity state
+}
+----
 
-Exceptions have the following effect:
+The `ManagedEntity` wrapper ensures that:
 
-* Any declared checked exception will be thrown if the command handler (or an interceptor) threw one of that type.
- If a checked exception is thrown that has not been declared, it is wrapped in a `CommandExecutionException`, which is a `RuntimeException`.
-* When a timeout occurs, the default behavior is to return `null` from the method.
- This can be changed by declaring a `TimeoutException`.
- If this exception is declared, a `TimeoutException` is thrown instead.
-* When a thread is interrupted while waiting for a result, the default behavior is to return null.
- In that case, the interrupted flag is set back on the thread.
- By declaring an `InterruptedException` on the method, this behavior is changed to throw that exception instead.
- The interrupt flag is removed when the exception is thrown, consistent with the java specification.
-* Other runtime exceptions may be declared on the method, but will not have any effect other than clarification to the API user.
+* Changes to entity state are tracked
+* State modifications are persisted when the processing context commits
+* The repository maintains consistent state throughout command processing
 
-Finally, there is the possibility to use annotations:
+=== Using repositories in command handlers
 
-* As specified in the parameter section, the `@MetaDataValue` annotation on a parameter will have the value of that parameter added as metadata value.
- The key of the metadata entry is provided as parameter to the annotation.
-* Methods annotated with `@Timeout` will block at most the indicated amount of time.
- This annotation is ignored if the method declares timeout parameters.
-* Classes annotated with `@Timeout` will cause all methods declared in that class to block at most the indicated amount of time, unless they are annotated with their own `@Timeout` annotation or specify timeout parameters.
+When command handlers are not on the entity itself, you use a repository to load and manage entities manually.
+This pattern is covered in xref:command-handlers.adoc#_command_centric_stateful_handlers[Command-centric stateful handlers].
 
 [source,java]
 ----
-public interface MyGateway {
+@Component
+public class OrderCommandHandler {
 
-    // fire and forget
-    void sendCommand(MyPayloadType command);
+    private final Repository<String, Order> orderRepository;
```

Comment:
Something for another PR too - it misses the entire `StateManager` pointer. Bit of a shame.